Flaky test: MinikubeWebhookInstallKT.shouldInstallAndInjectSidecar fails on verifyFailClosed

## Test Failure

The test `MinikubeWebhookInstallKT.shouldInstallAndInjectSidecar` failed on CI during the `verifyFailClosed` check.

**Failed Run:** https://github.com/kroxylicious/kroxylicious/actions/runs/26955049184/job/79534669897?pr=4067

**Error:**
```
java.lang.AssertionError: 

Expecting code to raise a throwable.
	at io.kroxylicious.kubernetes.webhook.AbstractWebhookInstallKT.verifyFailClosed(AbstractWebhookInstallKT.java:449)
	at io.kroxylicious.kubernetes.webhook.AbstractWebhookInstallKT.shouldInstallAndInjectSidecar(AbstractWebhookInstallKT.java:85)
```

## Expected Behavior

The test expects that when the webhook deployment is scaled to 0 replicas, attempting to create a pod should fail with a `KubernetesClientException` containing "sidecar-injector.kroxylicious.io" in the message, demonstrating the fail-closed behavior.

## Actual Behavior

The pod creation did not throw the expected exception, meaning the fail-closed mechanism did not engage as expected.

## Analysis

This appears to be a timing issue where:
1. The webhook is scaled to 0 replicas
2. The test immediately tries to create a pod
3. Kubernetes hasn't yet recognized the webhook as unavailable
4. The pod creation succeeds when it should fail

## Affected Test

`kroxylicious-kubernetes/kroxylicious-admission/src/test/java/io/kroxylicious/kubernetes/webhook/AbstractWebhookInstallKT.java:449`

## Impact

This is a flaky test that occasionally fails in CI. The fail-closed behavior itself is working correctly (as evidenced by other test runs passing), but the test has a race condition.

--- a/kroxylicious-kubernetes/kroxylicious-admission/pom.xml
+++ b/kroxylicious-kubernetes/kroxylicious-admission/pom.xml
@@ -180,6 +180,11 @@
             <scope>test</scope>
         </dependency>
         <dependency>
+            <groupId>org.awaitility</groupId>
+            <artifactId>awaitility</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-core</artifactId>
             <scope>test</scope>

--- a/kroxylicious-kubernetes/kroxylicious-admission/src/test/java/io/kroxylicious/kubernetes/webhook/AbstractWebhookInstallKT.java
+++ b/kroxylicious-kubernetes/kroxylicious-admission/src/test/java/io/kroxylicious/kubernetes/webhook/AbstractWebhookInstallKT.java
@@ -43,6 +43,7 @@ import io.kroxylicious.testing.integration.ShellUtils;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.assertj.core.api.Assumptions.assumeThat;
+import static org.awaitility.Awaitility.await;
 
 /**
  * Tests that the webhook can be installed from YAML manifests and that sidecar
@@ -406,29 +407,19 @@ abstract class AbstractWebhookInstallKT {
                 .withName("kroxylicious-webhook")
                 .scale(0);
 
-        client.apps().deployments()
-                .inNamespace(WEBHOOK_NS)
-                .withName("kroxylicious-webhook")
-                .waitUntilCondition(
-                        d -> d != null
-                                && d.getStatus() != null
-                                && (d.getStatus().getReadyReplicas() == null
-                                        || d.getStatus().getReadyReplicas() == 0),
-                        60, TimeUnit.SECONDS);
-
-        // Wait for the Service endpoints to be drained so the API server
-        // considers the webhook unreachable
-        client.endpoints()
-                .inNamespace(WEBHOOK_NS)
-                .withName("kroxylicious-webhook")
-                .waitUntilCondition(
-                        ep -> ep == null
-                                || ep.getSubsets() == null
-                                || ep.getSubsets().isEmpty()
-                                || ep.getSubsets().stream()
-                                        .allMatch(s -> s.getAddresses() == null
-                                                || s.getAddresses().isEmpty()),
-                        60, TimeUnit.SECONDS);
+        // Wait for webhook pods to actually terminate. The Kubernetes API server maintains
+        // persistent HTTP connection pools to webhook endpoints. Existing socket connections
+        // to terminating pods remain open and can continue servicing requests during the
+        // termination grace period (default 30s). The 60s timeout is chosen to be well-clear
+        // of the default terminationGracePeriodSeconds.
+        LOGGER.info("Waiting for webhook pods to terminate");
+        await().atMost(60, TimeUnit.SECONDS)
+                .until(() -> client.pods()
+                        .inNamespace(WEBHOOK_NS)
+                        .withLabel("app", "kroxylicious-webhook")
+                        .list()
+                        .getItems()
+                        .isEmpty());
 
         LOGGER.info("Webhook scaled to 0, attempting pod creation (expecting rejection)");
         var pod = new PodBuilder()

